### PR TITLE
[no ticket][risk=no] Puppeteer Simplify openSubtab function in Data page

### PR DESCRIPTION
**Pull request by aweng98**
Small improvement in openTab functions:
- If tab is already open (ariaSelected == true), then click on element is skipped.
- If open sub-tab in Data page, users do not have to remember to open Data tab first.

### CODE REVIEW

File: e2e/utils/waits-utils.ts [hunk 1]
```diff
@@ -39,7 +39,7 @@ export async function waitForDocumentTitle(page: Page, titleSubstr: string): Pro
     const jsHandle = await page.waitForFunction(t => {
       const actualTitle = document.title;
       return actualTitle.includes(t);
-    }, {}, titleSubstr);
+    }, {timeout: 30000}, titleSubstr);
```
Comment (aweng98) [inline e2e/utils/waits-utils.ts:42]: The default timeout for `waitForFunction` is 10 seconds. It is too short and made tests flaky. Increase it to 30 seconds helps to reduce flaky.

File: e2e/app/page/authenticated-page.ts [hunk 2]
```diff
@@ -38,11 +38,10 @@ export default abstract class AuthenticatedPage extends BasePage {
         this.isLoaded(),
       ]);
       return this;
-    } catch (e) {
+    } catch (err) {
       await savePageToFile(this.page);
       await takeScreenshot(this.page);
-      const title = await this.page.title();
-      throw new Error(`"${title}" page waitForLoad() encountered ${e}.`);
+      throw (err);
```
Comment (aweng98) [inline e2e/app/page/authenticated-page.ts:44]: `throw (err)` to keep original stacktrace unchanged. It was a mistake to do `throw new Error()` before.

File: e2e/app/page/workspace-base.ts [hunk 3]
```diff
@@ -94,6 +95,10 @@ export default abstract class WorkspaceBase extends AuthenticatedPage {
     const { waitPageChange = true } = opts;
     const selector = buildXPath({name: pageTabName, type: ElementType.Tab});
     const tab = new Link(this.page, selector);
+    const isSelected = await getPropValue<boolean>(await tab.asElementHandle(), 'ariaSelected');
```
Comment (jaycarlton) [inline e2e/app/page/workspace-base.ts:98]: aria as in Opera?

File: e2e/utils/waits-utils.ts (hunk 1 above)
Comment (jaycarlton) [reply e2e/utils/waits-utils.ts:42]: If you think individual steps are too slow, please file tickets on those, especially if it's a regression.

File: e2e/app/page/workspace-base.ts [hunk 4]
```diff
@@ -57,32 +58,32 @@ export default abstract class WorkspaceBase extends AuthenticatedPage {
    * Click Datasets subtab in Data page.
    * @param opts
    */
-  async openDatasetsSubtab(opts: {waitPageChange?: boolean} = {}): Promise<void> {
-    return this.openTab(TabLabels.Datasets, opts);
+  async openDatasetsSubtab(): Promise<void> {
+    return this.openDataPage().then(() => this.openTab(TabLabels.Datasets, {waitPageChange: false}));
```
Comment (jaycarlton) [inline e2e/app/page/workspace-base.ts:62]: Could you also use `await` here?

**Review by jaycarlton — APPROVED** [review]

File: e2e/app/page/workspace-base.ts (hunk 4 above)
Comment (aweng98) [reply e2e/app/page/workspace-base.ts:62]: `return` makes `await` redundant. same outcome with or without it.

**Review by aweng98 — COMMENTED** [review]
Thank you Jay

**Pull request merged**